### ISSUE TITLE
Improve S122: add configuration

### ISSUE THREAD

**Issue by DaFri-Nochiterov** [SonarSource/SonarTS#711]
**Rule key:** S122/no-statements-same-line

Would be nice if the rule above had a configuration allowing to disable it for some statements like this:

```json
  …
  "no-statements-same-line": [ true, {
    "break": true,
    "debugger": true,
    "do-while": true,
    "for": true,
    "if": true,
    "return": true,
    "var": true,
    // ^ done simplier (const, let, var):
    "variables": true
    …
  } ]
```

**Example**:

Configuration: `"no-statements-same-line": [ true, { break: false } ]`

```ts
…

// No error here:

function sayHello(target: Target) {
  let str = "Hello";

  switch (target) {
      case Target.World: str += ", our awesome world!"; break;
      case Target.Person: str += " to you, my dear friend!"; break;
      default: str = "An unexpected hello here :wave:"; break;
  }

  return str;
}

// But here:

function sayHello(target: Target) {
  let str = "Hello";
  if (target === Target.World) { str += ", out awesome world!"; return str; }
  …
}

// And here:

let a = "Hello"; let b = "World";
let helloWorld = `${a} {b}!`;

// So on...
```

**Comment by vilchik-elena** [comment]
IMO `case Target.World: str += ", our awesome world!"; break;` is unreadable and should not be ignored by the rule

I think disabling this particular lines (if you are so motivated to format this way) is a better option

**Comment by DaFri-Nochiterov** [comment]
Welp, it was just example I wrote for Issue.

Here's a real example from my code:
![image](https://user-images.githubusercontent.com/10401817/44848942-b360ed80-ac82-11e8-8025-f7a7372965cb.png)

For me, it's perfectly readable. And I don't really want to comment every line to disable the warning, nor extract value returning to function instead, just some customization about the rules like one in the first post of this Issue would be great. Maybe make it depending on line length or something 🤔

**Comment by vilchik-elena** [comment]
@DaFri-Nochiterov indeed your screenshot is readable, thanks to highlighting and homogeneous code. 
I would agree on making `break` inside `case` excluded based on config. But not every kind of statement

**Comment by DaFri-Nochiterov** [comment]
Yeah, at least this. Would be awesome, the only places I'm getting errors from this rule are switch-case lines where I don't really see line-break helpful to readability
